.github/workflows/deploy.yml

```diff
@@ -0,0 +1,28 @@
+name: CD
+
+on:
+  workflow_run:
+    workflows: ['CI']
+    types:
+      - completed
+    branches: [master]
+
+jobs:
+  deploy:
+    name: Github pages
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
```

Comment:
What do you think about putting whole workflow in single file and make that condition more readable?
```yaml
on: push

jobs:
  build:
    ...
  deploy:
    if: ${{ github.ref == 'master' }}
    needs: build
    ...
```

That way deploy should never run unless `build` will be completed sucessfully.

Reply:
The same result but in the single file. Currently if we would like to remove/change/add deploy stage we just remove `deploy` file and thats all. No corelation between `CI tasks` and deployment itself (expect deploy trigger conditions). Yes, your proposition looks less complicated (or rather it is little bit readable because of single file structure). I tried to imagine case when we would like to extend deployment process with eg. documentation in other tool + another place but I can't convince myself that it will ever happen 😅   No knowledge about that. Yea, lets keep it simple as it should :)